include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -334,50 +334,51 @@ struct transform_reduce
                 }
             }
         }
-        __local_mem[__item_id.get_local_id(0)] = __res_0;
+        return __res_0;
     }
 
 #else
-    template <typename _NDItemId, typename _Size, typename _AccLocal, typename... _Acc>
-    void
-    seq_impl(const _NDItemId __item_id, const _Size __n, const _Size __global_offset, const _AccLocal& __local_mem,
+    template <typename _NDItemId, typename _Size, typename... _Acc>
+    _Tp
+    seq_impl(const _NDItemId __item_id, const _Size __n, const _Size __global_offset,
              const _Acc&... __acc) const
     {
         auto __global_idx = __item_id.get_global_id(0);
         auto __local_idx = __item_id.get_local_id(0);
         const _Size __adjusted_global_id = __global_offset + __iters_per_work_item * __global_idx;
         const _Size __adjusted_n = __global_offset + __n;
-        // Add neighbour to the current __local_mem
         if (__adjusted_global_id + __iters_per_work_item < __adjusted_n)
         {
             // Keep these statements in the same scope to allow for better memory alignment
-            typename _AccLocal::value_type __res = __unary_op(__adjusted_global_id, __acc...);
+            _Tp __res = __unary_op(__adjusted_global_id, __acc...);
             _ONEDPL_PRAGMA_UNROLL
             for (_Size __i = 1; __i < __iters_per_work_item; ++__i)
                 __res = __binary_op(__res, __unary_op(__adjusted_global_id + __i, __acc...));
-            __local_mem[__local_idx] = __res;
+            return __res;
         }
         else if (__adjusted_global_id < __adjusted_n)
         {
             const _Size __items_to_process = __adjusted_n - __adjusted_global_id;
             // Keep these statements in the same scope to allow for better memory alignment
-            typename _AccLocal::value_type __res = __unary_op(__adjusted_global_id, __acc...);
+            _Tp __res = __unary_op(__adjusted_global_id, __acc...);
             for (_Size __i = 1; __i < __items_to_process; ++__i)
                 __res = __binary_op(__res, __unary_op(__adjusted_global_id + __i, __acc...));
-            __local_mem[__local_idx] = __res;
+            return __res;
         }
+        // TODO: find fix for types with no default constructor
```

Comment:
Since we've ascertained this doesn't affect any of our tests at present, I think we can leave this as a TODO for now.